api/wfx.go

```diff
@@ -0,0 +1,453 @@
+	status, err := status.Update(ctx, server.storage, request.Id, request.Body, eligible)
+	if err != nil {
+		switch ftag.Get(err) {
+		case ftag.NotFound:
+			return api.PutJobsIdStatus404JSONResponse(api.ErrorResponse{
+				Errors: &[]api.Error{JobNotFound},
+			}), nil
+		case ftag.InvalidArgument:
+			err2 := InvalidRequest
+			err2.Message = err.Error()
+			return api.PutJobsIdStatus400JSONResponse(api.ErrorResponse{
+				Errors: &[]api.Error{err2},
+			}), nil
+		default:
+			return nil, fault.Wrap(err)
+		}
+	}
+	if request.Params.XResponseFilter != nil {
+		return NewJQFilter(*request.Params.XResponseFilter, *status), nil
+	}
+	return api.PutJobsIdStatus200JSONResponse(*status), nil
+}
+
+func (server WfxServer) DeleteJobsIdTags(ctx context.Context, request api.DeleteJobsIdTagsRequestObject) (api.DeleteJobsIdTagsResponseObject, error) {
+	var body []string
+	if request.Body != nil {
+		body = *request.Body
+	}
+	tags, err := tags.Delete(ctx, server.storage, request.Id, body)
+	if err != nil {
+		if ftag.Get(err) == ftag.NotFound {
+			return api.DeleteJobsIdTags404JSONResponse(api.ErrorResponse{
+				Errors: &[]api.Error{JobNotFound},
+			}), nil
+		}
+		return nil, fault.Wrap(err)
+	}
+	if request.Params.XResponseFilter != nil {
+		return NewJQFilter(*request.Params.XResponseFilter, tags), nil
+	}
+	return api.DeleteJobsIdTags200JSONResponse(tags), nil
+}
+
+func (server WfxServer) GetJobsIdTags(ctx context.Context, request api.GetJobsIdTagsRequestObject) (api.GetJobsIdTagsResponseObject, error) {
+	tags, err := tags.Get(ctx, server.storage, request.Id)
+	if err != nil {
+		if ftag.Get(err) == ftag.NotFound {
+			return api.GetJobsIdTags404JSONResponse(api.ErrorResponse{
+				Errors: &[]api.Error{JobNotFound},
+			}), nil
+		}
+		return nil, fault.Wrap(err)
+	}
+	if request.Params.XResponseFilter != nil {
+		return NewJQFilter(*request.Params.XResponseFilter, tags), nil
+	}
+	return api.GetJobsIdTags200JSONResponse(tags), nil
+}
+
+func (server WfxServer) PostJobsIdTags(ctx context.Context, request api.PostJobsIdTagsRequestObject) (api.PostJobsIdTagsResponseObject, error) {
+	var body []string
+	if request.Body != nil {
+		body = *request.Body
+	}
+	tags, err := tags.Add(ctx, server.storage, request.Id, body)
+	if err != nil {
+		if ftag.Get(err) == ftag.NotFound {
+			return api.PostJobsIdTags404JSONResponse(api.ErrorResponse{
+				Errors: &[]api.Error{JobNotFound},
+			}), nil
+		}
+		return nil, fault.Wrap(err)
+	}
+	if request.Params.XResponseFilter != nil {
+		return NewJQFilter(*request.Params.XResponseFilter, tags), nil
+	}
+	return api.PostJobsIdTags200JSONResponse(tags), nil
+}
+
+func (server WfxServer) GetWorkflows(ctx context.Context, request api.GetWorkflowsRequestObject) (api.GetWorkflowsResponseObject, error) {
+	var offset int64
+	if request.Params.ParamOffset != nil {
+		offset = *request.Params.ParamOffset
+	}
+	var limit int32 = 10
```

Comment:
While `10` is a reasonable limit, it should be a "global" one to make it consistent with the same limit set above – also to `10`, wich is consistent, at least for now :)